components/claim-mgt/org.wso2.carbon.identity.claim.metadata.mgt/src/main/java/org/wso2/carbon/identity/claim/metadata/mgt/dao/CacheBackedClaimDialectDAO.java

```diff
@@ -48,21 +50,21 @@ public List<ClaimDialect> getClaimDialects(int tenantId) throws ClaimMetadataExc
         }
 
         claimDialectList = super.getClaimDialects(tenantId);
-        claimDialectCache.putClaimDialects(tenantId, claimDialectList);
+        claimDialectDAOCache.addToCacheIfNoDuplicate(tenantId, new ArrayList<>(claimDialectList), tenantId);
 
         if (log.isDebugEnabled()) {
             log.debug("Cache miss for claim dialect list for tenant: " + tenantId + ". Updated cache with claim " +
                     "dialect list retrieved from database. Claim dialect list size: " + claimDialectList.size());
         }
-        return claimDialectList;
+        return new ArrayList<>(claimDialectList);
```

Comment:
is this nessesary?

Reply:
This is to avoid corrupting the cache during merging as Java objects are passed by reference.

Once the cache returns the list, if any operations are performed on the returned object, the cached value gets affected even if we do not explicitly add the updated list to cache. Therefore, a new list would have to be created before merging in any case and handling it at the cache level is safer and more maintainable for the future changes as well.